Core/include/Acts/Geometry/BlueprintOptions.hpp

```diff
@@ -0,0 +1,30 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/NavigationPolicyFactory.hpp"
+
+#include <memory>
+
+namespace Acts {
+
+struct BlueprintOptions {
+  std::shared_ptr<NavigationPolicyFactory> defaultNavigationPolicyFactory{
+      makeDefaultNavigationPolicyFactory()};
+
+  void validate() const;
+
+  ~BlueprintOptions();
```

Comment:
do we need an explicit destructor? this will implicitly delete copy and move

Reply:
So, I was trying to make this compile with a forward declaration. Here, one trick I've found is that if the destructor is defined in a compilation unit, the `shared_ptr`'s destructor is not instantiated, and it will accept an incomplete type in the header.

It compiled locally just fine, but the missing include check still failed to compile it. I ended up just including the required header and dropped the forward declaration.

Consequently, I can also remove the destructor.